Use outlined version of thumb up/down icons in merchant feedback prompt snackbar for consistent visual language

Jinksi opened the pull request:
Fixes #10530

#### Changes proposed in this Pull Request

* Updates the imported version of `@wordpress/icons` from `8.2.3` to the latest version `10.19.0` so we can use the [newer outlined version of the thumbs up/down button](https://github.com/WordPress/gutenberg/blob/8b88ada2fd675509bf0c39a55c23a75cc67987cb/packages/icons/src/index.js#L278-L279)
* Uses outlined thumbs up/down icons in the merchant feedback snackbar prompt "Yes"/"No" buttons, to match the design (NgNHoJR3z8t8OwxJ5FshIY-fi-10_10162) and maintain a visual language consistent with WP-admin

<img width="424" alt="image" src="https://github.com/user-attachments/assets/3f3a792e-77e8-47e7-98e7-effd274a2eb7" />


#### 💭 Note on impact of `@wordpress/icons` library update

* After looking through the [`@wordpress/icons` changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/CHANGELOG.md), I don't see anything that should impact the rest of WooPayments.
* WooPayments doesn't import any of the changed icons since `8.2.3`, most of which are additions
* Snapshot tests were failing due to SVG changes, however, so I've manually checked the following:
	* Multi currency search icon in ✅
	* Payment details vertical more icon/button for refund menu ✅ 
	* Deposits overview card calendar icon ✅

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->


* Enable the dev feature flag `_wcpay_feature_prompt_merchant_for_review`.
	* `update_option( '_wcpay_feature_prompt_merchant_for_review', '1' )`
* Ensure your store meets the campaign eligibility criteria, or manually set `'wporgReview2025' => true` [here](https://github.com/Automattic/woocommerce-payments/blob/61fc121aec4d5718734f5a720517e694875c70de/includes/class-wc-payments-account.php#L378)
	- Lifetime TPV > $1,000 USD
	- 5+ completed payouts
	- Account in good standing (not suspended/rejected)
* Visit Payments → Overview.
* Ensure you see the new snackbar prompt.
* Ensure the Yes and No buttons use the outlined version of the icon.
* Optionally, run through various other WooPayments flows and check that icons are being displayed correctly.
	* See these [15 other files](https://github.com/search?q=repo%3AAutomattic%2Fwoocommerce-payments+%22from+%27%40wordpress%2Ficons%27%22&type=code) where we import `@wordpress/icons`.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️) **N/A**
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : **QA Testing Not Applicable** since eligibility is difficult to prepare, manual internal testing is suitable.
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. **Not a critical flow**
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. **What's new in 9.1.0 post: paJDYF-gXj-p2**

Jinksi committed:
Upgrade WordPress icons package to version 10.19.0 to access thumbsUp/Down icons

Jinksi committed:
Use outlined thumbsUp and thumbsDown icons in merchant feedback snackbar

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10531` or branch name `fix/10530-merchant-feedback-prompt-outlined-thumb-icons` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/10530-merchant-feedback-prompt-outlined-thumb-icons) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 038e3f9a6313b0ab0d46866c468332ff03405cc8
- Build time: 2025-03-10 23:57:55 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Jinksi committed:
Add changelog entries (separate one for `@wordpress/icons` update)

Jinksi committed:
Import `Icon` from `@wordpress/icons` to reduce import complexity

Jinksi committed:
package-lock.json update after running `npm install`

Jinksi committed:
update PaymentDetails snapshot tests after checking for visual bugs

Changes were to the moreVertical button https://github.com/WordPress/gutenberg/blob/8b88ada2fd675509bf0c39a55c23a75cc67987cb/packages/icons/src/library/more-vertical.js#L6

Jinksi committed:
Update Multi Currency snapshot tests after checking for visual bugs

Changes were to the search icon https://github.com/WordPress/gutenberg/blob/8b88ada2fd675509bf0c39a55c23a75cc67987cb/packages/icons/src/library/search.js#L6

Jinksi committed:
Update Deposits Overview snapshot tests after checking for visual bugs

Changes were to the calendar icon https://github.com/WordPress/gutenberg/blob/8b88ada2fd675509bf0c39a55c23a75cc67987cb/packages/icons/src/library/calendar.js#L6

brucealdridge reviewed (approved):
✅ Reviewed code.
✅ Visually checked a random sample of pages (Overview, Settings, Disputes List, Dispute Details, Dispute Response, Payout List, Payout Detail)
